lib/socket.js

```diff
@@ -62,20 +61,64 @@ class Transport extends EventEmitter {
   //
   _onSocketData(chunk) {
     const packets = [];
-    this._buffer += chunk;
+    let parsedLength;
+    let newPartLength;
+
+    if (this._buffer) {
+      newPartLength = chunk.indexOf(0);
+      if (newPartLength === -1) {
+        this._buffer = Buffer.concat(
```

Comment:
@lundibundi, from my observations, it is less common to receive long messages that will be divided more than once. And it is quite difficult to pick a starting size value for the buffer which will lead to improvement instead of being an unneeded overhead, I think it should be determined in an empirical way.
Also, my question in https://github.com/metarhia/jstp/pull/254#discussion_r127284438 was about this https://github.com/metarhia/jstp/pull/254#discussion_r127280400 (my bad, wasn't specific enough), meaning I was asking about the buffer growth: as far as I understand, it will lead to the same copying of the initial buffer that we have right now, and I don't understand how this will be better for performance. Can you, please, explain this part of your idea exactly?